warehouse/query-core/src/test/java/datawave/test/MacTestUtil.java

```diff
@@ -0,0 +1,166 @@
+        waitForPropertyRemoval(tops, tableName, Collections.singletonList(property));
+    }
+
+    /**
+     * When a property is removed from accumulo the test must wait until the change is persisted in ZooKeeper. Otherwise, test may execute with an incorrect set
+     * of assumptions.
+     *
+     * @param tops
+     *            an instance of {@link TableOperations}
+     * @param tableName
+     *            the table name
+     * @param properties
+     *            the list of removed properties
+     */
+    public static void waitForPropertyRemoval(TableOperations tops, String tableName, List<String> properties) {
+        try {
+            long start = System.currentTimeMillis();
+            boolean allRemoved = false;
+            while (!allRemoved) {
+                allRemoved = true;
+                Iterable<Map.Entry<String,String>> props = tops.getProperties(tableName);
+                for (Map.Entry<String,String> prop : props) {
+                    if (properties.contains(prop.getKey())) {
+                        allRemoved = false;
+                        break;
+                    }
+                }
+            }
+            long elapsed = System.currentTimeMillis() - start;
+            log.trace("removed {} properties in {} ms", properties.size(), elapsed);
+        } catch (AccumuloException | TableNotFoundException e) {
+            fail("Exception while verifying property removal");
+            throw new RuntimeException("Exception while verifying property removal", e);
+        }
+    }
+
+    /**
+     * When a property is added to accumulo the test must wait until the change is persisted in ZooKeeper. Otherwise, test may execute with an incorrect set of
+     * assumptions.
+     *
+     * @param tops
+     *            an instance of {@link TableOperations}
+     * @param tableName
+     *            the table name
+     * @param property
+     *            the property to remove
+     */
+    public static void waitForPropertyAddition(TableOperations tops, String tableName, String property) {
+        waitForPropertyAddition(tops, tableName, Collections.singletonList(property));
+    }
+
+    /**
+     * When a property is added to accumulo the test must wait until the change is persisted in ZooKeeper. Otherwise, test may execute with an incorrect set of
+     * assumptions.
+     *
+     * @param tops
+     *            an instance of {@link TableOperations}
+     * @param tableName
+     *            the table name
+     * @param properties
+     *            the list of removed properties
+     */
+    public static void waitForPropertyAddition(TableOperations tops, String tableName, List<String> properties) {
+        try {
+            long start = System.currentTimeMillis();
+            boolean allAdded = false;
+            while (!allAdded) {
+                List<String> additions = new ArrayList<>();
```

Comment:
If you used Set instead of List, you could just check for `additions.equals(properties)` without the sort calls.